Circuit/Components/VacuumTubes/Triode.cs

```diff
@@ -138,14 +140,23 @@ public override void Analyze(Analysis Mna)
             {
                 case TriodeModel.ChildLangmuir:
                     Expression Ed = Mu * Vgk + Vpk;
-                    ip = Call.If(Ed > 0, K * (Ed ^ 1.5), 0);
+                    ip = Call.If(Ed > 0, K * Binary.Power(Ed, 1.5), 0);
                     ig = 0;
                     ik = -(ip + ig);
                     break;
                 case TriodeModel.Koren:
                     Expression E1 = Ln1Exp(Kp * (1.0 / Mu + Vgk * Binary.Power(Kvb + Vpk * Vpk, -0.5))) * Vpk / Kp;
-                    ip = Call.If(E1 > 0, (E1 ^ Ex) / Kg, 0);
-                    ig = Call.Max(Vgk - Vg, 0) / Rgk;
+                    ip = Mna.AddUnknownEqualTo(Call.If(E1 > 0, 2d * (E1 ^ Ex) / Kg, 0));
+
+                    var vg = (double)Vg;
+                    var knee = (double)Kn;
+                    var rg1 = (double)Rgk;
```

Comment:
Same as above, I think casting to double is probably not what we want. Keeping them as `decimal` or `Real` would avoid the issues with `double` and make it so you don't need to bother with avoiding `Expression` operator overloads.

Reply:
Done